src/databaseTests/src/test/kotlin/org/vaccineimpact/api/databaseTests/tests/burdenEstimateRepository/PopulateStochasticBurdenEstimateSetTests.kt

```diff
@@ -0,0 +1,263 @@
+        for (i in 0..2)
+        {
+            assertThat(records[i][BURDEN_ESTIMATE.MODEL_RUN]).isEqualTo(expectedIds[0])
+        }
+        for (i in 3..5)
+        {
+            assertThat(records[i][BURDEN_ESTIMATE.MODEL_RUN]).isEqualTo(expectedIds[1])
+        }
+    }
+
+    private fun getEstimatesInOrder(db: JooqContext): Result<Record>
+    {
+        // We order the rows coming back so they are in a guaranteed order. This allows
+        // us to write simple hardcoded expectations.
+        val t = BURDEN_ESTIMATE_STOCHASTIC
+        return db.dsl.select(t.BURDEN_ESTIMATE_SET, t.COUNTRY, t.YEAR, t.AGE, t.VALUE, t.MODEL_RUN)
+                .select(BURDEN_OUTCOME.CODE)
+                .from(BURDEN_ESTIMATE_STOCHASTIC)
+                .join(BURDEN_OUTCOME)
+                .on(BURDEN_OUTCOME.ID.eq(BURDEN_ESTIMATE_STOCHASTIC.BURDEN_OUTCOME))
+                .orderBy(BURDEN_ESTIMATE_STOCHASTIC.COUNTRY, BURDEN_OUTCOME.CODE)
+                .fetch()
+    }
+
+    private fun checkRecord(record: Record, setId: Int,
+                            year: Int, age: Int, country: String, outcomeCode: String, outcomeValue: BigDecimal)
+    {
+        val t = BURDEN_ESTIMATE_STOCHASTIC
+        assertThat(record[t.BURDEN_ESTIMATE_SET]).isEqualTo(setId)
+        assertThat(record[t.COUNTRY]).isEqualTo(country)
+        assertThat(record[t.YEAR]).isEqualTo(year)
+        assertThat(record[t.AGE]).isEqualTo(age)
+        assertThat(record[BURDEN_OUTCOME.CODE]).isEqualTo(outcomeCode)
+        assertThat(record[t.VALUE]).isEqualTo(outcomeValue)
+    }
+
+    private fun data(runs: List<String?> = listOf(null, null)) = sequenceOf(
+            BurdenEstimateWithRunId(diseaseId, runs[0], 2000, 50, "AFG", "Afghanistan", 1000.toDecimal(), mapOf(
+                    "deaths" to 10.toDecimal(),
+                    "cases" to 100.toDecimal()
+            )),
+            BurdenEstimateWithRunId(diseaseId, runs[1], 1980, 30, "AGO", "Angola", 2000.toDecimal(), mapOf(
+                    "deaths" to 20.toDecimal(),
+                    "dalys" to 73.6.toDecimal()
+            ))
+    )
+
+    private fun addModelRuns(db: JooqContext, responsibilitySetId: Int, modelVersionId: Int): ModelRunTestData
+    {
+        val runs = listOf("marathon", "sprint")
+        val runParameterSetId = db.addModelRunParameterSet(responsibilitySetId, modelVersionId, username, "Test")
+        return ModelRunTestData(runParameterSetId, runs.map { runId ->
+            runId to db.addModelRun(runParameterSetId, runId)
+        })
+    }
+
+    private val defaultProperties = CreateBurdenEstimateSet(
+            BurdenEstimateSetType(BurdenEstimateSetTypeCode.STOCHASTIC),
+            modelRunParameterSetId = null
+    )
+
+    data class ModelRunTestData(val runParameterSetId: Int, val runs: List<Pair<String, Int>>)
+    {
+        val externalIds = runs.map { (runId, _) -> runId }
+        val internalIds = runs.map { (_, internalId) -> internalId }
+    }
+}
```

Comment:
This seems like a lot of code to duplicate. It's generally a good idea to keep tests as simple as possible, so I can see why you've done this but... maybe we should instead have one set of tests that we run twice, once for each type? e.g. by having an abstract base class for the tests, and then two test suites that just fill in the bits that differ?